src/crt/sllshl.src

```diff
@@ -0,0 +1,125 @@
+; Performs 48 bit ("short long long") bit shift left
+; 
+; Arguments:
+; ude = Most significant bytes of arg 1
+; uhl = Least significant bytes of arg 1
+; ubc = Amount to shift by (effectively just shift by c)
+
+
+; Returns:
+; ude:uhl = ude:uhl<<c
+
+assume	adl=1
+
+	section	.text
+	public	__sllshl
+__sllshl:
+	push iy
+	push de
+	push hl
+	pop iy ; iy will be used for least significant, hl for most going forward
```

Comment:
I think it would be less overhead overall if you don't juggle the registers, but instead do DE/HL swaps in the shifts, like:
```
add hl, hl
ex de, hl
adc hl, hl
ex de, hl
```
This is technically 1 more byte per single shift, but there should only be 7 single shifts total in the code (if the byte-aligned shifts are implemented via the stack), which is less overhead than the register juggling.